### PR TITLE
Create User first testing post

### PR DESCRIPTION
**Pull request by macklinu**
WIP, still need to fill out a couple of sections with basic examples.

### CODE REVIEW

File: src/content/posts/user-first-testing.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+## What about unit tests?
+
+Unit tests aren't bad, but I think they're often misused. When working alone or in a small team, I reach for unit tests in a couple of situations.
+
+1. I want fast feedback while I'm working with unfamiliar APIs or programming languages.
+1. I want to stress test a function with a lot of generated inputs to verify edge cases.
+
+### Fast feedback
+
+When I'm working with unfamiliar APIs or programming languages, I'll often write a unit test to verify that I'm using the API correctly. This allows me to explore something new and verify I understand the API or language well enough. This is even better with test runners that have a watch mode, like Jest or Vitest.
```
Comment (macklinu) [inline src/content/posts/user-first-testing.md:62]: Link to Jest and Vitest.

File: src/content/posts/user-first-testing.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+### Stress testing
+
+I tend to reach for a [property based][] testing framework like [fast-check][] when I want to test my functions in a way where I can explore edge cases and a variety of different inputs in a general way.
+
+Insert example here
```
Comment (macklinu) [inline src/content/posts/user-first-testing.md:68]: TODO

File: src/content/posts/user-first-testing.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Conclusion: recap why user first testing is good
```
Comment (macklinu) [inline src/content/posts/user-first-testing.md:70]: TODO
